src/api/exec.rs

```diff
@@ -120,13 +122,27 @@ impl Exec {
                 .map_err(|e| crate::conn::Error::Any(Box::new(e)))?,
         );
 
-        self.podman.post_upgrade_stream(ep, payload).await.map(|x| {
-            if self.is_tty {
-                tty::Multiplexer::new(x, tty::decode_raw)
+        let detach = opts.params.get("Detach").map(|value| value.as_bool().unwrap_or(false)).unwrap_or(false);
```

Comment:
This could potentially be slightly improved like this:
```suggestion
        let detach = opts.params.get("Detach").and_then(|value| value.as_bool()).unwrap_or(false);
```